datasource/http/data.go

```diff
@@ -62,6 +66,25 @@ func (d *Datasource) Configure(raws ...interface{}) error {
 			fmt.Errorf("the `url` must be specified"))
 	}
 
+	// Default to GET if no method is specified
+	if d.config.Method == "" {
+		d.config.Method = "GET"
+	}
+
+	// Check if the input is in the list of allowed methods
+	validMethod := false
+	allowedMethods := []string{"HEAD", "GET", "POST", "PUT"}
+	for _, method := range allowedMethods {
+		if method == d.config.Method {
+			validMethod = true
```

Comment:
I would consider adding a `break` here to exit early if the method is valid, very minor nit tbh